src/constants.ts

```diff
@@ -7,6 +7,18 @@ import { ProofOfEfficiency__factory } from "src/types/contracts/proof-of-efficie
 import { ProviderError } from "src/adapters/error";
 import { Chain, Currency, EthereumChain, EthereumChainId, Token, ZkEVMChain } from "src/domain";
 
+export const DAI_PERMIT_TYPEHASH =
+  "0xea2aa0a1be11a07ed86d755c93467f4f82362b452371d1ba94d1715123511acb";
+
+export const EIP_2612_PERMIT_TYPEHASH =
+  "0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9";
+
+export const EIP_2612_STANDARD_DOMAIN_TYPEHASH =
```

Comment:
The point is that this "standard" is not always 100% followed, so we end up having 2 flavors within the `EIP_2612` `DOMAIN_TYPEHASH`: _standard_ and _uniswap_.

Does it make sense?

Reply:
Yep, in fact that's why I asked. Because, you have the standard type hash, which is `EIP_2612_DOMAIN_TYPEHASH`, and then for any kind of variation (like the UNI token one), you have: `EIP_2612_UNI_DOMAIN_TYPEHASH` as it's a variation of the standard one. But I think that it's redundant to add the `STANDARD` keyword to the standard one as the EIP already indicates that it's the standard one.

Reply:
I get the point, but then I think we should remove `EIP_2612` from `EIP_2612_UNISWAP_DOMAIN_TYPEHASH` since this one is not standard.

I mean, if the reasoning is that `EIP_2612_STANDARD_DOMAIN_TYPEHASH` should not include `STANDARD` because `EIP_2612` implies that it follows the standard, then since uniswap does not "completely" follow the standard, we should name it something like `UNISWAP_DOMAIN_TYPEHASH`.

So we end up having: `EIP_2612_DOMAIN_TYPEHASH` and `UNISWAP_DOMAIN_TYPEHASH`.

Agreed?

Reply:
Mmm makes sense yep, I didn't see that. Okay, taking that into account this last proposal looks good yep!

Reply:
Well and of course the type becomes:

```typescript
export enum Permit {
  DAI = "DAI",
  EIP_2612 = "EIP_2612",
  UNISWAP = "UNISWAP",
}
```

Reply:
Done